### PR TITLE
Fixed cyclomatic complexity of get function

### PR DESCRIPTION
**Pull request by ArtemkaK1**
Splitted get function in 3 functions to fix it cyclomatic complexity from C to B and A

### CODE REVIEW

File: libs/langchain/langchain/vectorstores/vearch.py [hunk 1]
```diff
@@ -522,6 +522,30 @@ def delete(
...
+        content = ""
+        meta_info = {}
+        for field in docs_detail:
+            if field == "text":
+                content = docs_detail[field]
+                continue
+            elif field == "metadata":
+                meta_info["source"] = docs_detail[field]
+                continue
+        return content, meta_info
```
Comment (smttsp) [inline libs/langchain/langchain/vectorstores/vearch.py:547]: These two functions are exactly the same. The only difference is their inputs

```suggestion
    def fill_meta_info_and_content(self, input_dict):
        content = ""
        meta_info = {}
        if "text" in input_dict:
            content = input_dict[field]
        if "metadata" in input_dict:
            meta_info["source"] = input_dict[field]
        return content, meta_info
```

Your function calls will be 

```
content, meta_info = self.fill_meta_info_and_content(record["_source"])
```

and

```
content, meta_info = self.fill_meta_info_and_content(docs_detail)
```

respectively.

Comment (smttsp) [reply libs/langchain/langchain/vectorstores/vearch.py:547]: you can further improve this function by:

```suggestion
    def fill_meta_info_and_content(self, input_dict):
        content = input_dict.get("text", "")
        meta_info = {"source": input_dict["metadata"]} if "metadata" in input_dict else {}
        return content, meta_info
```

**Review by smttsp — COMMENTED** [review]
I think it is great you are splitting these functions.

However, as someone who has very limited knowledge about the repo, I would recommend:

1. It might be a good idea to add docstring. 
2. all the `continue` statements in the for loops are redundant. 
3. It might be better to try and reuse the function `fill_meta_info_and_content` for both conditions. I gave two possible ways to simplify the function that you might wanna consider. 

Please let me know if these are of any help,

**Comment by baskaryan** [comment]
Apologies for the slow review! Pr has some merge conflicts, happy to re-review if you'd like to resolve

**Pull request closed without merging**